src/fmeval/eval_algorithms/summarization_accuracy.py

```diff
@@ -1,159 +1,149 @@
+                f"Invalid rouge_type: {self.rouge_type} requested in SummarizationAccuracyConfig. "
+                f"Please choose from acceptable values: {ROUGE_TYPES}."
             )
 
-        if not BertscoreHelperModelTypes.model_is_allowed(self.model_type_for_bertscore):
+        if not BertscoreModelTypes.model_is_allowed(self.model_type_for_bertscore):
             raise EvalAlgorithmClientError(
                 f"Invalid model_type_for_bertscore: {self.model_type_for_bertscore} requested in "
-                f"SummarizationAccuracyConfig, please choose from acceptable values: {BertscoreHelperModelTypes.model_list()}"
+                f"SummarizationAccuracyConfig. Please choose from acceptable values: "
+                f"{BertscoreModelTypes.model_list()}."
             )
 
 
 class SummarizationAccuracy(EvalAlgorithmInterface):
-    """
-    Summarization Accuracy Eval algorithm
+    """Summarization Accuracy evaluation algorithm.
 
-    The aim of this eval algo is to evaluate how well a model can summarize text.
-    The algo uses a reference summary to compare the output generated by the model and a series
+    The aim of this evaluation algorithm is to evaluate how well a model can summarize text.
+    The algorithm uses a reference summary to compare the output generated by the model and a series
     of quality metrics based on overlapping between words (ROUGE and METEOR) and similarity scores (BERTScore).
     """
 
-    def __init__(self, eval_algorithm_config: SummarizationAccuracyConfig = SummarizationAccuracyConfig()):
-        """Default constructor
+    eval_name = EvalAlgorithm.SUMMARIZATION_ACCURACY.value
 
-        :param eval_algorithm_config: Summarization Accuracy eval algorithm config.
+    def __init__(
+        self, eval_algorithm_config: SummarizationAccuracyConfig = SummarizationAccuracyConfig(), use_ray: bool = True
+    ):
+        """SummarizationAccuracy initializer.
+
+        :param eval_algorithm_config: Summarization Accuracy evaluation algorithm config.
+        :param use_ray: Whether to use the Ray distributed computing framework to run the evaluation
+            algorithm logic. While using Ray will typically speed up the evaluation, setting this flag
+            to False can be useful for debugging purposes or in situations where computational resources
+            are limited.
         """
         super().__init__(eval_algorithm_config)
-        self.eval_name = EvalAlgorithm.SUMMARIZATION_ACCURACY.value
-        self._eval_algorithm_config = eval_algorithm_config
-        self._load_eval_helpers()
-        self._score_eval_func_mapping = {
-            METEOR_SCORE: get_meteor_score,
-            ROUGE_SCORE: get_rouge_score,
-            BERT_SCORE: get_bert_score,
-        }
+        meteor_transform = MeteorScore(
+            input_keys=[DatasetColumns.TARGET_OUTPUT.value.name, DatasetColumns.MODEL_OUTPUT.value.name],
+            output_keys=[METEOR_SCORE],
```

Comment:
can we have these defaults baked in the class rather than carried over here?

Reply:
Also, would be nice to have keys take both string and sequences as inputs (and convert to sequences internally) for better usability if there is only a key

Reply:
Will do.